### PR TITLE
MM-13525 Fix alignment of jewel on channel drawer icon

### PR DESCRIPTION
**Pull request by sudheerDev**
#### Summary
Fixes the mention and unread jewel alignment on IOS tablets.
 
Badge component and unread components needs to be wrapped with a `</view>`  to keep styles relative to each other

#### Ticket Link
[MM-13525](https://mattermost.atlassian.net/browse/MM-13525)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Has UI changes

#### Device Information
This PR was tested on: [Mobile and tablets emulators of both android and IOS] 

#### Screenshots

screenshot 2018-12-21 at 2 32 53 pm
screenshot 2018-12-21 at 2 32 38 pm
screenshot 2018-12-21 at 2 28 23 pm
screenshot 2018-12-21 at 2 28 15 pm
screenshot 2018-12-21 at 2 19 51 pm
screenshot 2018-12-21 at 2 19 39 pm
screenshot 2018-12-21 at 2 15 38 pm
screenshot 2018-12-21 at 2 15 29 pm

### CODE REVIEW

File: app/screens/channel/channel_nav_bar/channel_drawer_button.js [hunk 1]
```diff
@@ -133,8 +132,10 @@ class ChannelDrawerButton extends PureComponent {
                 style={style.container}
             >
                 <View style={[style.wrapper, {opacity: this.state.opacity}]}>
-                    {icon}
-                    {badge}
+                    <View>
```
Comment (enahum) [inline app/screens/channel/channel_nav_bar/channel_drawer_button.js]: is this `<View>` needed?

Comment (sudheerDev) [reply app/screens/channel/channel_nav_bar/channel_drawer_button.js]: Yup the view is needed to wrap the styles for `position absolute` other wise they move based on the available space hence the issue

**Comment by sudheerDev** [comment]
Dot position is always that way for tablets. I made an attempt to fix the dot position as well in this PR. It is really tricky as it is font based and not icon. 

screenshot 2018-12-24 at 9 15 21 pm
screenshot 2018-12-24 at 8 35 30 pm

**Comment by enahum** [comment]
Build failing

**Comment by sudheerDev** [comment]
@enahum Fixed the build issue

**Comment by jespino** [comment]
About the dot, maybe we can use something more "reliable" about padding/margin than a font. Maybe for the dot we can use a super-simple inlined SVG with a circle.

**Comment by enahum** [comment]
SVg is too slow, a png would be way better

**Comment by sudheerDev** [comment]
@enahum @jespino PNG can result in pixelation at higher density screens 
I used a  `<View/>` and made the  `.` here as it is simple to achieve it. It is consistent now and i was able to remove platform specific styles. 

@esethna 👇 
screenshot 2019-01-22 at 2 52 09 am
screenshot 2019-01-22 at 2 52 55 am
screenshot 2019-01-22 at 2 53 17 am
screenshot 2019-01-22 at 3 01 15 am
screenshot 2019-01-22 at 3 01 45 am
screenshot 2019-01-22 at 3 02 35 am

**Comment by enahum** [comment]
Again go with png in normal, @2x and @3x to avoid pixelation and the tint color can be applied without performance degradation

**Comment by sudheerDev** [comment]
> Again go with png in normal, @2x and @3X to avoid pixelation and the tint color can be applied without performance degradation

I don't see a good reason for doing it here. Same behaviour can be achieved without the png. We will be increasing the build by few kbs with out any added benefits. so, i would refrain from doing it. 


@esethna it is off by 0.5px or something i did attempt to fix it and it as close to center as possible now. i did trail and error with 1px in all directions but it is always off by very little. We might have to change the dimensions of outer circle to keep the dot at center. I would say we go ahead with this and create another ticket after having a look at it on a real devices about the position to see if we have to spend mana on this.

jesus or elias suggestions are about making it consistent across all platforms and devices and we have the consistency now(i took a similar approach to jesus suggestion). If we have a bug in positioning it will be for all devices and for both IOS and android so, it is more obvious and easy to fix.

**Comment by sudheerDev** [comment]
@enahum Just to be on the same page. You meant the dot alone to be png or the whole badge? 
We support theming for the badge that means we need to add different backgrounds to both the elements so we would have to use two images and which again brings to the same problem if i am not wrong.

**Comment by enahum** [comment]
@sudheerDev I was talking about the dot only

**Comment by sudheerDev** [comment]
@esethna Ok gave another shot to centre align the dot. Feel free to create another ticket for centring if this does not work for you. For my eyes now it seems it is aligned to the centre 

screenshot 2019-01-25 at 3 23 39 pm
screenshot 2019-01-25 at 3 31 12 pm

**Comment by enahum** [comment]
@sudheerDev build failure

**Comment by sudheerDev** [comment]
I am good with the changes here. Can we go ahead with the PR here?

**Pull request merged**